src/ReflectionParameter.php

```diff
@@ -132,23 +138,30 @@ public function __debugInfo(): array
      *
      * @return string
      */
-    public function __toString()
+    public function __toString(): string
     {
         $parameterType   = $this->getType();
         $isOptional      = $this->isOptional();
         $hasDefaultValue = $this->isDefaultValueAvailable();
         $defaultValue    = '';
         if ($hasDefaultValue) {
             $defaultValue = $this->getDefaultValue();
-            if (is_string($defaultValue) && strlen($defaultValue) > 15) {
-                $defaultValue = substr($defaultValue, 0, 15) . '...';
-            }
-            /* @see https://3v4l.org/DJOEb for behaviour changes */
-            if (is_float($defaultValue) && fmod($defaultValue, 1.0) === 0.0) {
-                $defaultValue = (int)$defaultValue;
-            }
 
-            $defaultValue = str_replace('\\\\', '\\', var_export($defaultValue, true));
+            if ($this->parameterNode->default instanceof Array_) {
+                $this->parameterNode->default->setAttribute('kind', Array_::KIND_SHORT);
+                $printer = new Standard(['shortArraySyntax' => true]);
+                $defaultValue = $printer->prettyPrintExpr($this->parameterNode->default);
```

Comment:
Note to myself: using Printer should pass the transformed default value on `___toString()`, eg; transformed array with magic const: `[___FUNCTION___ . "test"]` which should equivalent to `[namespaced\functiontest]`

but that may need separate PR